Production_scripts/convert_gp_costs.R

```diff
@@ -0,0 +1,143 @@
+#####################################################
+# Costs - GP Out of Hours
+# Author: Catherine Holland
+# Date: February 2022
+# Written on RStudio Server
+# Version of R - 3.6.1
+# Input -
+# Description -
+#####################################################
+
+
+library(dplyr)
+library(ggplot2)
+
+
+## Make a copy of the existing file ##
+# read data in
+current_file <- haven::read_sav(get_gp_ooh_costs_path())
+
+
+# write data to folder
+# .zsav
+haven::write_sav(current_file,
+  paste0(get_slf_dir(), "/Costs/Cost_GPOoH_Lookup_pre", latest_update(), ".zsav"),
+  compress = TRUE
+)
+
+
+
+## update file ##
+
+## Attendances taken from 2018 Primary Care Out of Hours Report ##
+# https://publichealthscotland.scot/publications/out-of-hours-primary-care-services-in-scotland/
+
+## Costs taken from R520 (Costbook) report for 2015/16 ##
+# https://beta.isdscotland.org/topics/finance/costs/ (R520)
+
+## The above should be checked / added to the Excel file 'OOH_Costs.xlsx' before running this syntax.
+
+
+
+## data ##
+gp_out_of_hours <- readxl::read_xlsx(paste0(get_slf_dir(), "/Costs/OOH_Costs.xlsx"))
+
+
+## data - wide to long ##
+gp_out_of_hours <-
+  gp_out_of_hours %>%
+  pivot_longer(c(ends_with("_Consultations"), ends_with("_Cost")),
+    names_to = c("year", ".value"),
+    names_pattern = "(\\d{4})_(.+)"
+  )
+
+## create cost per consultation ##
+gp_out_of_hours <-
+  gp_out_of_hours %>%
+  mutate(
+    cost_per_consultation = Cost * 1000 / Consultations
+  )
+
+
+## add in years by copying the most recent year ##
+latest_year <- 1920
```

Comment:
I think to move this to the top. And maybe rename `latest_cost_year` (I know I suggested the previous name too 😝